central/processlisteningonport/datastore/datastore_impl.go

```diff
@@ -12,23 +12,54 @@ import (
 	"github.com/stackrox/rox/central/processlisteningonport/store"
 	"github.com/stackrox/rox/generated/storage"
 	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/postgres"
 	"github.com/stackrox/rox/pkg/process/id"
 	"github.com/stackrox/rox/pkg/sac"
 	"github.com/stackrox/rox/pkg/sac/resources"
 	"github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/sync"
 	"github.com/stackrox/rox/pkg/uuid"
 )
 
 type datastoreImpl struct {
 	storage            store.Store
 	indicatorDataStore processIndicatorStore.DataStore
+	mutex              sync.RWMutex
 }
 
 var (
 	plopSAC = sac.ForResource(resources.DeploymentExtension)
 	log     = logging.LoggerForModule()
 )
 
+const (
```

Comment:
nit:  maybe pull the const out into a `pruning.go` in this package to reduce clutter